Packages/com.unity.inputsystem/InputSystem/Editor/ControlPicker/InputControlPickerDropdown.cs

```diff
@@ -140,86 +355,191 @@ private void AddDeviceTreeItem(InputControlLayout layout, AdvancedDropdownItem p
                 }
 
                 if (!matchesAnyInDeviceFilter)
-                    return;
+                    return false;
             }
 
-            if (m_Mode != InputControlPicker.Mode.PickDevice)
-                AddControlTreeItemsRecursive(layout, deviceItem, "", layout.name, null);
+            return true;
+        }
 
-            if (deviceItem.children.Any() || m_Mode == InputControlPicker.Mode.PickDevice)
-                parent.AddChild(deviceItem);
+        private void StartListening()
+        {
+            if (m_RebindingOperation == null)
+                m_RebindingOperation = new InputActionRebindingExtensions.RebindingOperation();
 
-            if (m_Mode != InputControlPicker.Mode.PickDevice)
-                foreach (var commonUsage in layout.commonUsages)
-                {
-                    var commonUsageGroup = new DeviceTreeViewItem(layout, commonUsage);
-                    AddControlTreeItemsRecursive(layout, commonUsageGroup, "", layout.name, commonUsage);
-                    if (commonUsageGroup.children.Any())
-                        parent.AddChild(commonUsageGroup);
-                }
+            m_RebindingOperation
+                .WithExpectedControlLayout(m_ExpectedControlLayout)
+                // Require minimum actuation of 0.15f. This is after deadzoning has been applied.
+                .WithMagnitudeHavingToBeGreaterThan(0.15f)
+                ////REVIEW: should we exclude only the system's active pointing device?
+                // With the mouse operating the UI, its cursor control is too fickle a thing to
+                // bind to. Ignore mouse position and delta and clicks.
+                // NOTE: We go for all types of pointers here, not just mice.
+                .WithControlsExcluding("<Pointer>/position")
+                .WithControlsExcluding("<Pointer>/delta")
+                .WithControlsExcluding("<Pointer>/button")
+                .WithControlsExcluding("<Mouse>/leftButton")
+                .WithControlsExcluding("<Mouse>/scroll")
```

Comment:
On my Mac, this will exclude the generic Mouse device from showing up in listening queries, but mouse interactions will still show up from the more specific HID device implementing my Touchpad. I think this is wrong. I don't think you would ever want to bind to the specific HID implementation of the MacBook touchpad. So we should hide that as well (somehow?).

Reply:
Yes, just not here IMO. Getting rid of undesirable HIDs should happen at a system-wide level. ATM we don't yet have good mechanisms for reigning in the fishnet-like behavior of the HID layout builder.